src/apis/filter/axios.ts

```diff
@@ -1,38 +1,74 @@
+import instance from '@apis/instance';
 import MESSAGES from '@apis/messages';
 import { isAxiosError } from 'axios';
 
-export const fetchFilteredList = async (
-  filterData: FilterType & { price: PriceType; content: string },
-  page: number,
-  userId?: string,
-) => {
-  const axiosInstance = getAxiosInstance();
-
+// v2 API
+export const fetchFilteredListV2 = async (params: TemplestaySearchParamsV2) => {
   try {
-    const response = await axiosInstance.post(`/search?page=${page}&userId=${userId}`, {
-      ...filterData,
-    });
+    const response = await instance.get('/v2/api/templestay', { params });
 
-    return response.data;
+    return response.data.data;
   } catch (error) {
     if (isAxiosError(error)) throw error;
     else throw new Error(MESSAGES.UNKNOWN_ERROR);
   }
 };
 
-export const fetchFilteredCount = async (
-  filterData: FilterType & { price: PriceType; content: string },
-) => {
-  try {
-    const response = await instance.post('/public/filter/count', {
-      ...filterData,
-    });
+// 필터 데이터를 v2 API 파라미터로 변환하는 헬퍼 함수
+export const convertToV2Params = (
+  groupedFilters: FilterType,
+  price: PriceType,
+  search: string,
+  page: number,
+  userId?: string,
+  sort?: string,
+): TemplestaySearchParamsV2 => {
+  const params: TemplestaySearchParamsV2 = {
+    page,
+    search: search && search.trim() !== '' ? search : undefined,
+    min: price.minPrice > 0 ? price.minPrice : undefined,
+    max: price.maxPrice < 30 ? price.maxPrice : undefined,
+    sort: sort && sort.trim() !== '' ? sort : undefined,
+    userId: userId && userId.trim() !== '' ? userId : undefined,
+  };
 
-    return response.data;
-  } catch (error) {
-    if (isAxiosError(error)) throw error;
-    else throw new Error(MESSAGES.UNKNOWN_ERROR);
+  // 각 필터 그룹의 선택된 아이템들을 콤마로 구분된 문자열로 변환
+  if (groupedFilters.region) {
+    const selectedRegions = Object.entries(groupedFilters.region)
+      .filter(([, isSelected]) => isSelected)
+      .map(([region]) => region);
+    if (selectedRegions.length > 0) {
+      params.region = selectedRegions.join(',');
+    }
+  }
+
+  if (groupedFilters.type) {
+    const selectedTypes = Object.entries(groupedFilters.type)
+      .filter(([, isSelected]) => isSelected)
+      .map(([type]) => type);
+    if (selectedTypes.length > 0) {
+      params.type = selectedTypes.join(',');
+    }
+  }
+
+  if (groupedFilters.activity) {
+    const selectedActivities = Object.entries(groupedFilters.activity)
+      .filter(([, isSelected]) => isSelected)
+      .map(([activity]) => activity);
+    if (selectedActivities.length > 0) {
+      params.activity = selectedActivities.join(',');
+    }
+  }
+
+  if (groupedFilters.etc) {
+    const selectedEtc = Object.entries(groupedFilters.etc)
+      .filter(([, isSelected]) => isSelected)
+      .map(([etc]) => etc);
+    if (selectedEtc.length > 0) {
+      params.etc = selectedEtc.join(',');
+    }
```

Comment:
각 필터 그룹에 대해 동일한 패턴의 코드가 반복되고 있어, 해당 부분을 별도의 헬퍼 함수로 분리한 후 이를 활용해 각 필터 값을 추출하는 방식으로 리팩토링하면 좋을 것 같습니다 !

```suggestion
  const getSelectedItems = (filterGroup?: Record<string, number>): string | undefined => {
    if (!filterGroup) return undefined;
    const selected = Object.entries(filterGroup)
      .filter(([, value]) => value === 1)
      .map(([item]) => item);
    return selected.length > 0 ? selected.join(',') : undefined;
  };

  params.region = getSelectedItems(groupedFilters.region);
  params.type = getSelectedItems(groupedFilters.type);
  params.activity = getSelectedItems(groupedFilters.activity);
  params.etc = getSelectedItems(groupedFilters.etc);

```